dvc/output/base.py

```diff
@@ -288,10 +290,11 @@ def save(self):
 
         self.hash_info = self.get_hash()
 
-    def commit(self):
+    def commit(self, filter_info=None):
         if not self.exists:
             raise self.DoesNotExistError(self)
-
+        if filter_info:
+            self.hash_info = self.get_hash(filter_info)
         assert self.hash_info
         if self.use_cache:
             self.cache.save(self.path_info, self.cache.tree, self.hash_info)
```

Comment:
Now `self.path_info` doesn't correspond to `self.hash_info`, as `self.hash_info` is for `filter_info` and not for `self.path_info` :slightly_frowning_face:

Reply:
You could do something like
```suggestion
            self.cache.save(self.path_info, self.cache.tree, self.hash_info, filter_info=filter_info)
```
and let the cache properly filter `hash_info` on itsown (that way we will avoid computing the hash twice as you currently do in `get_hash(filter_info)`).

Or you could
```
self.cache.save(filter_info, self.cache.tree, hash_info)
```
where hash_info is self.hash_info but filtered according to `filter_info`. Notice that you don't have to do a full `get_hash` call for `filter_info`, you could just filter the existing `self.hash_info` according to `filter_info`.

Reply:
Doene I choose to perform decision on output/base.py layer